src/copy-paste.js

```diff
@@ -0,0 +1,347 @@
+define([
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/tsv',
+    'vellum/core'
+], function (
+    $,
+    _,
+    mugs,
+    tsv
+) {
+    var PREAMBLE = ["Form Builder clip", "version 1"],
+        vellum,
+        offScreen = {top: -10000, left: -10000},
+        hiddenTextarea = $('<textarea></textarea>').css({
+            position: 'absolute',
+            width: 0,
+            height: 0
+        }).css(offScreen).appendTo('body');
+
+    function focusTextarea($focus, value) {
+        if ($focus.length === 0) {
+            $focus = $("body");
+        }
+        hiddenTextarea.css({top: $focus.offset().top});
+        hiddenTextarea.val(value);
+        hiddenTextarea.focus();
+        hiddenTextarea.select();
+    }
+
+    function unfocusTextarea($focus, clear) {
+        $focus.focus();
+        var value = hiddenTextarea.val();
+        if (clear && value) {
+            // HACK fix intermittent multiple-paste on Chrome (timing related?)
+            hiddenTextarea.val("");
+        }
+        return value;
+    }
+
+    function onCut(opts) {
+        var $focus = $(':focus');
+        if ($focus.is('.jstree-anchor')) {
+            var text = opts.cut();
+            if (text) {
+                focusTextarea($focus, text);
+                setTimeout(function () {
+                    unfocusTextarea($focus);
+                }, 10);
+            }
+        }
+    }
+
+    function onCopy(opts) {
+        var $focus = $(':focus');
+        if ($focus.is('.jstree-anchor')) {
+            var text = opts.copy();
+            if (text) {
+                focusTextarea($focus, text);
+                setTimeout(function () {
+                    unfocusTextarea($focus);
+                }, 10);
+            }
+        }
+    }
+
+    function onPaste(opts) {
+        var $focus = $(':focus');
+        if ($focus.length === 0 || $focus.parents('.fd-tree').length ||
+                                   $focus.is(hiddenTextarea)) {
+            focusTextarea($focus);
+            setTimeout(function () {
+                var pasteValue = unfocusTextarea($focus, true);
+                // on chrome this gets called twice,
+                // the first time with a blank value
+                if (pasteValue) {
+                    var errors = opts.paste(pasteValue);
+                    if (errors.length) {
+                        vellum._resetMessages([{
+                            level: "parse-warning",
```

Comment:
wasn't sure if you wanted to keep using constants, but thought i'd flag. (another benefit of the constant is if another developer makes modifications, they know what levels to choose from)